### ISSUE TITLE
Not getting Tweets

### ISSUE THREAD

**Issue by Spanishwalker** [digitalmethodsinitiative/dmi-tcat#154]
Hi,
I am using the Debian one command line installation on Amazon EC2 since this morning. I made some queries, but they are not storing any tweet.

I don't know if it needs to be refreshed somehow, or there is actually a problem somewhere.

**Comment by mtsguim** [comment]
Hi guys,
I'm using Ubuntu version with automated installation. The application says it has collected 16,000, 20,000 tweets, but when I do the analysis download files find no tweet line. As well as fellow up I do queries, run the code, the program says ta collecting but are not storing any tweet.

Do you have any idea what is this our problem?

Hugs,

Marco Túlio Silva Guimarães

**Comment by dentoir** [comment]
Hi @Spanishwalker 

Can you login to your EC2 server and execute the following command

`tail -n 5 /etc/crontab`

Please send us the output. Can you also make a screenshot of how your capture/ page looks and attach it?

**Comment by dentoir** [comment]
Hi @mtsguim 

Please open a separate question for a separate issue. Could you attach a screenshot of your capture/ page and also a screenshot of your analysis/ page.

Before exporting tweets you should make a selection (a query _inside_ on of your tweet bins) which contains tweets.

**Comment by Spanishwalker** [comment]
Hi @dentoir 
![captura de pantalla 2016-02-23 a les 11 12 25](https://cloud.githubusercontent.com/assets/12252768/13249285/b9fe10e8-da23-11e5-813e-91bf4882a464.png)


ok, So here is what I have done step by step.

1. I launched an instance with debian 8.0
2. I connected through SSH Java midterm
3. I installed DMI. I guess everything right, following all default options. I selected the option for only DMI server purposes during the installation
4. Since I could not have access to eui/capture either with publicIP/capture, I had to change the web server ports. I went to Security Groups, under Networking on the Amazon EC2, and added HTTP Inbound. Also I selected Source anywhere.
5. Now I can have access by publicIP/capture. I start a new hashtag to get data, but does not get any.
6. I log in again with SSH to run the command you told me. This is what it shows, not sure if there is the place to run the command:

Run DMI-TCAT URL expander every hour
0 *     * * *   jaruizso   (cd /var/www/dmi-tcat/helpers; sh urlexpand.sh)

# Run TCAT controller every minute
* *     * * *   jaruizso   (cd /var/www/dmi-tcat/capture/stream/; php controller.php)

**Comment by dentoir** [comment]
Hi @Spanishwalker 

Thank you for the information. It all looks correct so far. You did edit the file /var/www/dmi-tcat/config.php and insert your Twitter API keys, right? Without those API keys there can be no capture.

Could you maybe login to your server and send us the output of the following command:

`tail -n 50 /var/www/dmi-tcat/logs/track.error.log`

**Comment by Spanishwalker** [comment]
I did insert the API keys when the installation. I followed the one line command installation on Debian.

It says: 

tail -n 50 /var/www/dmi-tcat/logs/track.error.log               
tail: cannot open â/var/www/dmi-tcat/logs/track.error.logâ for reading: No such file or directory


Houston, we have a problem.

**Comment by dentoir** [comment]
Okay, can you try running the following command (this should start the tweet tracking process in the foreground on your terminal. you can interrupt it with CTRL-C) and send us the output:

`cd /var/www/dmi-tcat/capture/stream/; php dmitcat_track.php`

**Comment by Spanishwalker** [comment]
Hi again, and many thanks for the help. Actually there is nothing to interrupt, when I run the command:

admin@ip-172-31-26-254:~$ cd /var/www/dmi-tcat/capture/stream/; php dmitcat_track.php
PHP Warning:  touch(): Unable to create file /var/www/dmi-tcat/proc/track.lock because Permission denied in /var/www/dmi-tcat/capture/common/functions.php on line 428
PHP Warning:  fopen(/var/www/dmi-tcat/proc/track.lock): failed to open stream: No such file or directory in /var/www/dmi-tcat/capture/common/functions.php on line 430
PHP Warning:  flock() expects parameter 1 to be resource, boolean given in /var/www/dmi-tcat/capture/common/functions.php on line 432
PHP Warning:  fclose() expects parameter 1 to be resource, boolean given in /var/www/dmi-tcat/capture/common/functions.php on line 444
PHP Warning:  file_put_contents(/var/www/dmi-tcat/logs/track.error.log): failed to open stream: Permission denied in /var/www/dmi-tcat/capture/common/functions.php on line 454

**Comment by dentoir** [comment]
Hi @Spanishwalker 

Thank you. Could you maybe run the following two commands?

Command 1:

`ls -la /var/www/dmi-tcat/`

Command 2 (this is running the tweet capture again, but this time under the correct user):

`cd /var/www/dmi-tcat/capture/stream/; sudo -u  jaruizso php dmitcat_track.php`

**Comment by Spanishwalker** [comment]
Hi there. Here is the result of running both commands :)

admin@ip-172-31-26-254:~$ ls -la /var/www/dmi-tcat/
total 80
drwxr-xr-x 10 root root  4096 Feb 23 10:38 .
drwxr-xr-x  4 root root  4096 Feb 23 10:35 ..
drwxr-xr-x  6 root root  4096 Feb 23 10:35 analysis
drwxr-xr-x  9 root root  4096 Feb 23 10:35 capture
drwxr-xr-x  2 root root  4096 Feb 23 10:35 common
-rw-r--r--  1 root root  5866 Feb 23 10:38 config.php
-rw-r--r--  1 root root  5697 Feb 23 10:35 config.php.example
drwxr-xr-x  8 root root  4096 Feb 23 10:35 .git
-rw-r--r--  1 root root    79 Feb 23 10:35 .gitignore
drwxr-xr-x  2 root root  4096 Feb 23 10:35 helpers
drwxr-xr-x  2 root root  4096 Feb 23 10:35 import
-rw-r--r--  1 root root 10273 Feb 23 10:35 LICENSE
drwxr-xr-x  2 root root  4096 Feb 23 10:35 logs
drwxr-xr-x  2 root root  4096 Feb 23 10:35 proc
-rw-r--r--  1 root root   201 Feb 23 10:35 .project
-rw-r--r--  1 root root   298 Feb 23 10:35 README.md
admin@ip-172-31-26-254:~$ cd /var/www/dmi-tcat/capture/stream/; sudo -u jaruizso php dmitcat_track.php
sudo: unknown user: jaruizso
sudo: unable to initialize policy plugin


What username is referring to?

**Comment by dentoir** [comment]
Hi @Spanishwalker 

Is it possible that you entered username 'jaruizso' during the auto-installation process, when asked "Shell username for TCAT ownership ...."?

It looks as though you entered that username while that username did not exist on the system. That is why you cannot capture any tweets now. It's not running jobs for a non-existent user.

Can you add the user 'jaruizso'? Use the command below (run as root), choose a password and just press enter on all the other questions:

`adduser jaruizso`

After this command, you need to fix ownerships to the TCAT directory. Execute the following commands (as root):

```
chown -R jaruizso:jaruizso /var/www/dmi-tcat/
cd /var/www/dmi-tcat
chown www-data:www-data analysis/cache
chmod 755 analysis/cache
chown jaruizso:jaruizso logs proc
chmod 755 logs
chmod 755 proc
```

Hope this helps.

Cheers,

Emile

**Comment by Spanishwalker** [comment]
I think we kind of solved it. Let me explain.

I run the commands and that was the outcome:

admin@ip-172-31-26-254:~$ sudo su
root@ip-172-31-26-254:/home/admin# adduser jaruizso
Adding user `jaruizso' ...
Adding new group `jaruizso' (1001) ...
Adding new user `jaruizso' (1001) with group `jaruizso' ...
Creating home directory `/home/jaruizso' ...
Copying files from `/etc/skel' ...
Enter new UNIX password: 
Retype new UNIX password: 
passwd: password updated successfully
Changing the user information for jaruizso
Enter the new value, or press ENTER for the default
        Full Name []: 
        Room Number []: 
        Work Phone []: 
        Home Phone []: 
        Other []: 
Is the information correct? [Y/n] y
root@ip-172-31-26-254:/home/admin# chown -R jaruizso:jaruizso /var/www/dmi-tcat/root@ip-172-31-26-254:/home/admin# cd /var/www/dmi-tcat
root@ip-172-31-26-254:/var/www/dmi-tcat# chown www-data:www-data analysis/cache
root@ip-172-31-26-254:/var/www/dmi-tcat# chmod 755 analysis/cache
root@ip-172-31-26-254:/var/www/dmi-tcat# chown jaruizso:jaruizso logs proc
root@ip-172-31-26-254:/var/www/dmi-tcat# chmod 755 logs
root@ip-172-31-26-254:/var/www/dmi-tcat# chmod 755 proc
root@ip-172-31-26-254:/var/www/dmi-tcat# 


I waited some minutes to see if things were moving. Since data was not gathered yet, I run the two previous commands you game me. And that was the outcome:

root@ip-172-31-26-254:/home/admin# ls -la /var/www/dmi-tcat/
total 80
drwxr-xr-x 10 jaruizso jaruizso  4096 Feb 23 10:38 .
drwxr-xr-x  4 root     root      4096 Feb 23 10:35 ..
drwxr-xr-x  6 jaruizso jaruizso  4096 Feb 23 10:35 analysis
drwxr-xr-x  9 jaruizso jaruizso  4096 Feb 23 10:35 capture
drwxr-xr-x  2 jaruizso jaruizso  4096 Feb 23 10:35 common
-rw-r--r--  1 jaruizso jaruizso  5866 Feb 23 10:38 config.php
-rw-r--r--  1 jaruizso jaruizso  5697 Feb 23 10:35 config.php.example
drwxr-xr-x  8 jaruizso jaruizso  4096 Feb 23 10:35 .git
-rw-r--r--  1 jaruizso jaruizso    79 Feb 23 10:35 .gitignore
drwxr-xr-x  2 jaruizso jaruizso  4096 Feb 23 10:35 helpers
drwxr-xr-x  2 jaruizso jaruizso  4096 Feb 23 10:35 import
-rw-r--r--  1 jaruizso jaruizso 10273 Feb 23 10:35 LICENSE
drwxr-xr-x  2 jaruizso jaruizso  4096 Feb 23 10:35 logs
drwxr-xr-x  2 jaruizso jaruizso  4096 Feb 23 10:35 proc
-rw-r--r--  1 jaruizso jaruizso   201 Feb 23 10:35 .project
-rw-r--r--  1 jaruizso jaruizso   298 Feb 23 10:35 README.md
root@ip-172-31-26-254:/home/admin# cd /var/www/dmi-tcat/capture/stream/; sudo -u jaruizso php dmitcat_track.php
PHP Deprecated:  mysql_connect(): The mysql extension is deprecated and will be removed in the future: use mysqli or PDO instead in /var/www/dmi-tcat/common/functions.php on line 5


Now it seems gathering data, and seems everything working ok now, But I am not sure of the last message. Its all right now?

**Comment by Spanishwalker** [comment]
ok, I confirm that the data gathering has stopped. I run the command tail -n 50 /var/www/dmi-tcat/logs/track.error.log and thats what it says:

admin@ip-172-31-26-254:~$ tail -n 50 /var/www/dmi-tcat/logs/track.error.log
2016-02-26 13:24:25     running php version 5.6.17-0+deb8u1 in mode cli with extensions Core,date,ereg,libxml,openssl,pcre,zlib,bcmath,bz2,calendar,ctype,dba,dom,hash,fileinfo,filter,ftp,gettext,SPL,iconv,mbstring,pcntl,session,posix,Reflection,standard,shmop,SimpleXML,soap,sockets,Phar,exif,sysvmsg,sysvsem,sysvshm,tokenizer,wddx,xml,xmlreader,xmlwriter,zip,PDO,curl,geos,json,mysql,mysqli,pdo_mysql,readline,mhash,Zend OPcache (ini file: /etc/php5/cli/php.ini)
2016-02-26 13:24:25     installing term signal handler for this script
2016-02-26 13:24:25     geoPHP library is fully functional
2016-02-26 13:24:25     started script track with pid 846
2016-02-26 13:24:25     connecting to API socket
2016-02-26 13:24:25     connecting - query array (
  'track' => 'brexit,TTIP',

**Comment by dentoir** [comment]
Can you try:

`cd /var/www/dmi-tcat/capture/stream/; sudo -u jaruizso php controller.php`

That should start tracking in the background (which is what the cronjob is supposed to do).

**Comment by dentoir** [comment]
And send me the output of the logs/track.error.log and logs/controller.log after a minute of running this

**Comment by Spanishwalker** [comment]
I run the first command, and I see data is starting again. Not sure if it will stop one time I close the connection to the instance. Lets see.


**First command:**

root@ip-172-31-26-254:/var/www/dmi-tcat/capture/stream# tail -n 50 /var/www/dmi-tcat/logs/track.error.log                   
2016-02-26 13:24:25     running php version 5.6.17-0+deb8u1 in mode cli with extensions Core,date,ereg,libxml,openssl,pcre,zlib,bcmath,bz2,calendar,ctype,dba,dom,hash,fileinfo,filter,ftp,gettext,SPL,iconv,mbstring,pcntl,session,posix,Reflection,standard,shmop,SimpleXML,soap,sockets,Phar,exif,sysvmsg,sysvsem,sysvshm,tokenizer,wddx,xml,xmlreader,xmlwriter,zip,PDO,curl,geos,json,mysql,mysqli,pdo_mysql,readline,mhash,Zend OPcache (ini file: /etc/php5/cli/php.ini)
2016-02-26 13:24:25     installing term signal handler for this script
2016-02-26 13:24:25     geoPHP library is fully functional
2016-02-26 13:24:25     started script track with pid 846
2016-02-26 13:24:25     connecting to API socket
2016-02-26 13:24:25     connecting - query array (
  'track' => 'brexit,TTIP',
)
2016-02-26 17:01:19     running php version 5.6.17-0+deb8u1 in mode cli with extensions Core,date,ereg,libxml,openssl,pcre,zlib,bcmath,bz2,calendar,ctype,dba,dom,hash,fileinfo,filter,ftp,gettext,SPL,iconv,mbstring,pcntl,session,posix,Reflection,standard,shmop,SimpleXML,soap,sockets,Phar,exif,sysvmsg,sysvsem,sysvshm,tokenizer,wddx,xml,xmlreader,xmlwriter,zip,PDO,curl,geos,json,mysql,mysqli,pdo_mysql,readline,mhash,Zend OPcache (ini file: /etc/php5/cli/php.ini)
2016-02-26 17:01:19     installing term signal handler for this script
2016-02-26 17:01:19     geoPHP library is fully functional
2016-02-26 17:01:19     started script track with pid 1148
2016-02-26 17:01:19     connecting to API socket
2016-02-26 17:01:19     connecting - query array (
  'track' => 'brexit,TTIP,draghi,soteu,Schengen,eurovision',

**Second command:**

root@ip-172-31-26-254:/var/www/dmi-tcat/capture/stream# tail -n 50 /var/www/dmi-tcat/logs/controller.log                    
2016-02-26 17:01:18     received instruction to execute 'reload' for script track
2016-02-26 17:01:18     received instruction to execute 'reload' for script track
2016-02-26 17:01:18     received instruction to execute 'reload' for script track
2016-02-26 17:01:18     received instruction to execute 'reload' for script track
2016-02-26 17:01:18     received instruction to execute 'reload' for script track
2016-02-26 17:01:18     received instruction to execute 'reload' for script track
2016-02-26 17:01:18     script track was not running - starting

**Comment by Spanishwalker** [comment]
ok, for now it seems still working fine...gathering tweets

**Comment by dentoir** [comment]
Hi @Spanishwalker,

Glad to hear it is working now, if you have no more problems with TCAT, I'll close this ticket.

**Comment by Spanishwalker** [comment]
Hi @dentoir,

I have been checking during the weekend, and for some reason it stops time to time. I have to run the following command to start the gathering again:

cd /var/www/dmi-tcat/capture/stream/; sudo -u jaruizso php controller.php 

As far as I know, I did not hit any rate limit. At the beginning I thought it was because of that, so I installed DMI Tcat in another server, with another Twitter App to decrease the possibility of rate limit. But the problem persists. In one server (the original, with 5 keywords), or in the new (with one keyword but with lot of data).
